nsot/models.py

```diff
@@ -858,115 +856,78 @@ def get_next_network(self, prefix_length, num=None, as_objects=True):
 
         cidr = self.ip_network
 
-        # If this network is an interconnect network, generate the hosts as
-        # network objects, otherwise just subnet based on the prefix_length.
-        # This is so that .0 and .1 could be allocated from a /31, for example.
-        if cidr.prefixlen in settings.NETWORK_INTERCONNECT_PREFIXES:
-            log.debug('CIDR %s is an interconnect!', cidr)
-            subnets = (ipaddress.ip_network(ip) for ip in cidr)
-        else:
-            log.debug('cidr %s is NOT an interconnect!', cidr)
-            subnets = cidr.subnets(new_prefix=prefix_length)
-
-        # Exclude children that are in busy states.
-        children = self.get_descendants()
-
-        # Partition children into busy children and all children, storing
-        # the `.ip_network` property up front.
-        busy_children = [
-            c.ip_network for c in children if (
-                c.state in self.BUSY_STATES
-            )
-        ]
-        children = [
-            c.ip_network for c in children
-        ]
-
-        # Prepopulate children_seen with networks that have the same prefix
-        # length as wanted `prefix_length`.
-        children_seen = [
-            c for c in children if (
-                c.prefixlen == prefix_length
-            )
-        ]
-
-        # Dirty subnets cannot be used. Pre-seed w/ busy children.
-        dirty = [c for c in busy_children if (c.prefixlen == prefix_length)]
-
-        # Pre-filter any pre-seeded dirty subnets as a generator.
-        subnets = (s for s in subnets if s not in dirty)
-
-        # Subnets we want.
-        wanted = []
-
-        # Keep iterating until we've found the number of networks we want.
-        while len(wanted) < num:
+        if prefix_length > cidr.max_prefixlen:
             try:
-                next_subnet = next(subnets)
+                next(cidr.subnets(new_prefix = prefix_length))
             except ValueError as err:
                 raise exc.ValidationError({'prefix_length': err.message})
-            except StopIteration:
-                break
 
-            log.debug('>> CHILDREN_SEEN: %s items, DIRTY: %s items',
-                      len(children_seen), len(dirty))
+        if strict:
+            children = [c.ip_network for c in self.get_children()]
+        else:
+            children = [c.ip_network for c in self.get_descendants() 
+                            if c.prefix_length >= prefix_length]
+
+        exclude_nums = {}
+        
+        network_prefix = cidr.network_address 
+        # get all numbers formed by bits in child network address
+        # that do not exist in parent network address
+        a = int(network_prefix) >> (cidr.max_prefixlen - prefix_length)
+        for c in children:
+            b = int(c.network_address) >> (cidr.max_prefixlen - prefix_length)
+            exclude_nums[a ^ b] = c.prefixlen
```

Comment:
For various children, it looks like `exclude_nums[a ^ b]` could get overwritten with different values of `c.prefixlen`. For example if `children` consists of:
```
10.1.1.1/32
10.1.1.2/32
10.1.2.0/24
10.1.2.1/32
```

... then the prefix length stored would be `32`. With how `exclude_nums` is used later, wouldn't you want to store the smallest prefix length instead of the last one seen?

From my understanding of this PR I don't think this breaks anything, but it's some possibly unintended behavior that if fixed would yield better performance.